joy2twist/src/joy2twist_node.cpp

```diff
@@ -0,0 +1,86 @@
+#include "joy2twist/joy2twist_node.hpp"
+
+namespace joy2twist
+{
+Joy2TwistNode::Joy2TwistNode(std::shared_ptr<ros::NodeHandle> nh) : nh_(std::move(nh))
+{
+  load_parameters();
+
+  joy_sub_ = nh_->subscribe("/joy", 1, &Joy2TwistNode::joy_cb, this);
+  twist_pub_ = nh_->advertise<MsgTwist>("/cmd_vel", 1);
+}
+
+void Joy2TwistNode::load_parameters()
+{
+  nh_->param(
+    "/joy2twist_node/linear_velocity_factor/fast", linear_velocity_factors_[FAST], defaults::VELOCITY_FACTORS.at(FAST));
+  nh_->param(
+    "/joy2twist_node/linear_velocity_factor/regular", linear_velocity_factors_[REGULAR], defaults::VELOCITY_FACTORS.at(REGULAR));
+  nh_->param(
+    "/joy2twist_node/linear_velocity_factor/slow", linear_velocity_factors_[SLOW], defaults::VELOCITY_FACTORS.at(SLOW));
+  
+    nh_->param(
+    "/joy2twist_node/angular_velocity_factor/fast", angular_velocity_factors_[FAST], defaults::VELOCITY_FACTORS.at(FAST));
+    nh_->param(
+    "/joy2twist_node/angular_velocity_factor/regular", angular_velocity_factors_[REGULAR], defaults::VELOCITY_FACTORS.at(REGULAR));
+    nh_->param(
+    "/joy2twist_node/angular_velocity_factor/slow", angular_velocity_factors_[SLOW], defaults::VELOCITY_FACTORS.at(SLOW));
+
+  nh_->param(
+    "/joy2twist_node/angular_velocity_factor", angular_velocity_factors_,
+    defaults::VELOCITY_FACTORS);
+  
+  nh_->param(
+    "/joy2twist_node/button_index_map/axis/angular_z", button_index_.angular_z,
+    defaults::ANGULAR_Z);
+  nh_->param(
+    "/joy2twist_node/button_index_map/axis/linear_x", button_index_.linear_x, defaults::LINEAR_X);
+  nh_->param(
+    "/joy2twist_node/button_index_map/axis/linear_y", button_index_.linear_y, defaults::LINEAR_Y);
+  nh_->param(
+    "/joy2twist_node/button_index_map/dead_man_switch", button_index_.dead_man_switch,
+    defaults::DEAD_MAN_SWITCH);
+  nh_->param(
+    "/joy2twist_node/button_index_map/fast_mode", button_index_.fast_mode, defaults::FAST_MODE);
+  nh_->param(
+    "/joy2twist_node/button_index_map/slow_mode", button_index_.slow_mode, defaults::SLOW_MODE);
```

Comment:
i think that using global namespaces is not a good idea, it will be better to use private node handle (`ros::NodeHandle ph("~")`) and just reference parameter names.

Reply:
:heavy_check_mark:

Reply:
maybe we can use `ph_` instead of `p_nh_`

Reply:
:heavy_check_mark: